patroni/validator.py

```diff
@@ -91,6 +116,8 @@ def validate_host_port(host_port: str, listen: bool = False, multiple_hosts: boo
         for host in hosts:
             # Check if "socket.IF_INET" or "socket.IF_INET6" is being used and instantiate a socket with the identified
             # protocol
+            if not is_valid_host(host):
+                raise ConfigParseError("{} is not a valid host address".format(host))
             proto = socket.getaddrinfo(host, "", 0, socket.SOCK_STREAM, 0, socket.AI_PASSIVE)
```

Comment:
Meh, `socket.getaddrinfo()` is supposed to raise an exception:
```python
In [2]: socket.getaddrinfo('1244.0.0.1', "", 0, socket.SOCK_STREAM, 0, socket.AI_PASSIVE)
---------------------------------------------------------------------------
gaierror                                  Traceback (most recent call last)
<ipython-input-5-9e9cff7106dc> in <module>
----> 1 socket.getaddrinfo('1244.0.0.1', "", 0, socket.SOCK_STREAM, 0, socket.AI_PASSIVE)

/usr/lib/python3.10/socket.py in getaddrinfo(host, port, family, type, proto, flags)
    953     # and socket type values to enum constants.
    954     addrlist = []
--> 955     for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
    956         af, socktype, proto, canonname, sa = res
    957         addrlist.append((_intenum_converter(af, AddressFamily),

gaierror: [Errno -2] Name or service not known
```

If it doesn't happen for you - something is wrong with your environment.

I don't see the immediate need to introduce `is_valid_host()` function. You can just mock `socket.getaddrinfo()` instead, that will return/raise expected results.

Reply:
I think I've find the reason, when I have my clash(a tunnel proxy) opened, getaddrinfo gave
me an address different from the input:

```
>>> socket.getaddrinfo('1244.0.0.1', "", 0, socket.SOCK_STREAM, 0, socket.AI_PASSIVE)
[(<AddressFamily.AF_INET: 2>, <SocketKind.SOCK_STREAM: 1>, 6, '', ('198.18.4.188', 0))]
```

but after I close clash, getaddrinfo raise an exception as expected:

```
>>> socket.getaddrinfo('1244.0.0.1', "", 0, socket.SOCK_STREAM, 0, socket.AI_PASSIVE)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/Cellar/python@3.11/3.11.5/Frameworks/Python.framework/Versions/3.11/lib/python3.11/socket.py", line 962, in getaddrinfo
    for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
socket.gaierror: [Errno 8] nodename nor servname provided, or not known
```

I will just shut down clash when I run the unit test, thanks.

Reply:
Adding a proper mock still makes sense.

Reply:
> Adding a proper mock still makes sense.

I will close this PR and add the mock in another pull request.